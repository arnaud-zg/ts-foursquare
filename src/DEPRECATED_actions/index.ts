import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
