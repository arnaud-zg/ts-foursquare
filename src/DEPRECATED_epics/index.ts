export * from './life'
export * from './lists'
export * from './photos'
export * from './venues'
