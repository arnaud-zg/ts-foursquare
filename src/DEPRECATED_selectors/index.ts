export * from './life'
export * from './photos'
export * from './request'
export * from './status'
export * from './venues'
