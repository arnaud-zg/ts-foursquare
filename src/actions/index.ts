import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
