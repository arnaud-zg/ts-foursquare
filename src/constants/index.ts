export const PROJECT_NAME = 'ts-foursquare'
