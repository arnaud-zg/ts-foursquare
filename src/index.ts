require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
