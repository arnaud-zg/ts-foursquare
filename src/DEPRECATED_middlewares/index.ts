import { combineEpics, createEpicMiddleware } from 'redux-observable'
import { NStore } from '../../types'
import { TRootAction } from '../DEPRECATED_actions'
import {
  getAccessTokenEpic,
  getPhotosDetailsEpic,
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesListedEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic,
} from '../DEPRECATED_epics'

export const rootEpic = combineEpics(
  getAccessTokenEpic,
  getPhotosDetailsEpic,
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesListedEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic
)
export const epicMiddleware = createEpicMiddleware<
  TRootAction,
  TRootAction,
  NStore.IState
>()
