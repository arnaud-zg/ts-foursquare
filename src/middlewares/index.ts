import { combineEpics, createEpicMiddleware } from 'redux-observable'
import { NStore } from '../../types/store.d'
import { TRootAction } from '../actions'
import {
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic,
} from '../epics'

export const rootEpic = combineEpics(
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic
)
export const epicMiddleware = createEpicMiddleware<
  TRootAction,
  TRootAction,
  NStore.IState
>()
