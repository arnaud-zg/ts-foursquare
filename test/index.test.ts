import * as tsFoursquare from '../src'

describe('tsFoursquare', () => {
  it('should have a list of method', () => {
    expect(tsFoursquare).toMatchSnapshot()
  })
})
