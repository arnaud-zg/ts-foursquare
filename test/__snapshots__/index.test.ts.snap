// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsFoursquare should have a list of method 1`] = `
Object {
  "ASYNC_ACTION_NAME_MAPPING": Object {
    "GET_VENUES_EXPLORE_CANCEL": "getVenuesExploreAsync",
    "GET_VENUES_EXPLORE_FAILURE": "getVenuesExploreAsync",
    "GET_VENUES_EXPLORE_REQUEST": "getVenuesExploreAsync",
    "GET_VENUES_EXPLORE_SUCCESS": "getVenuesExploreAsync",
    "GET_VENUES_LIKES_CANCEL": "getVenuesLikesAsync",
    "GET_VENUES_LIKES_FAILURE": "getVenuesLikesAsync",
    "GET_VENUES_LIKES_REQUEST": "getVenuesLikesAsync",
    "GET_VENUES_LIKES_SUCCESS": "getVenuesLikesAsync",
    "GET_VENUES_SEARCH_CANCEL": "getVenuesSearchAsync",
    "GET_VENUES_SEARCH_FAILURE": "getVenuesSearchAsync",
    "GET_VENUES_SEARCH_REQUEST": "getVenuesSearchAsync",
    "GET_VENUES_SEARCH_SUCCESS": "getVenuesSearchAsync",
    "GET_VENUES_TRENDING_CANCEL": "getVenuesTrendingAsync",
    "GET_VENUES_TRENDING_FAILURE": "getVenuesTrendingAsync",
    "GET_VENUES_TRENDING_REQUEST": "getVenuesTrendingAsync",
    "GET_VENUES_TRENDING_SUCCESS": "getVenuesTrendingAsync",
  },
  "EIconSize": Object {
    "L": "64",
    "M": "44",
    "S": "32",
    "XL": "88",
  },
  "StandaloneStore": [Function],
  "configureStore": [Function],
  "credentialsSelector": [Function],
  "getIconUrlList": [Function],
  "getLocationHref": [Function],
  "getLocationSearch": [Function],
  "getVenuesExploreAsync": Object {
    "cancel": [Function],
    "failure": [Function],
    "request": [Function],
    "success": [Function],
  },
  "getVenuesLikesAsync": Object {
    "cancel": [Function],
    "failure": [Function],
    "request": [Function],
    "success": [Function],
  },
  "getVenuesSearchAsync": Object {
    "cancel": [Function],
    "failure": [Function],
    "request": [Function],
    "success": [Function],
  },
  "getVenuesSuggestCompletionAsync": Object {
    "cancel": [Function],
    "failure": [Function],
    "request": [Function],
    "success": [Function],
  },
  "getVenuesTrendingAsync": Object {
    "cancel": [Function],
    "failure": [Function],
    "request": [Function],
    "success": [Function],
  },
  "initialState": Object {
    "life": Object {
      "credentials": Object {
        "clientId": "",
        "clientSecret": "",
      },
      "status": false,
    },
    "status": Object {},
    "venues": Object {
      "entities": Object {},
      "recommendedPlaces": Array [],
      "trendingEntities": Object {},
    },
  },
  "lifeActions": Object {
    "putCredentials": [Function],
  },
  "lifeSelector": [Function],
  "lifeStatusSelector": [Function],
  "moduleStateSelector": [Function],
  "putCredentials": [Function],
  "statusSelector": [Function],
  "venueCategoriesStateSelector": [Function],
  "venueLocationSelector": [Function],
  "venueSelector": [Function],
  "venuesActions": Object {
    "getVenuesExploreAsync": Object {
      "cancel": [Function],
      "failure": [Function],
      "request": [Function],
      "success": [Function],
    },
    "getVenuesLikesAsync": Object {
      "cancel": [Function],
      "failure": [Function],
      "request": [Function],
      "success": [Function],
    },
    "getVenuesSearchAsync": Object {
      "cancel": [Function],
      "failure": [Function],
      "request": [Function],
      "success": [Function],
    },
    "getVenuesSuggestCompletionAsync": Object {
      "cancel": [Function],
      "failure": [Function],
      "request": [Function],
      "success": [Function],
    },
    "getVenuesTrendingAsync": Object {
      "cancel": [Function],
      "failure": [Function],
      "request": [Function],
      "success": [Function],
    },
  },
  "venuesSelector": [Function],
}
`;
